target/linux/ath79/dts/qca9558_librerouter-v1.dts

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca9557.dtsi"
+
+/ {
+	compatible = "librerouter,librerouter-v1", "qca,qca9557";
+	model = "LibreRouter v1";
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	aliases {
+		led-boot = &system;
+		led-failsafe = &system;
+		led-running = &system;
+		led-upgrade = &system;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		system: system {
+			label = "librerouter:green:system";
+			gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "heartbeat";
+		};
+
+		wifi_green {
+			label = "librerouter:green:wlan2g";
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+	};
+
+	gpio-export {
```

Comment:
Would it be possible to replace the gpio-export construct with the userspace gpio-switch equivalent?

(If you need a example device, please look at: fa3c2676ab0ce5fbb648249da31e31bbc61a3e3d )

Reply:
Why is gpio-switch prefered over gpio-export? 
Do you know if there is a "named  way" to use the gpio with gpio-switch? or to use it it's necesary to first do something like `uci get system.poe_passthrough.gpio_pin` to know the number of the gpio and then use it through `/sys/class/gpio/gpio1/`?

Reply:
I did the requested change, but still want to know why it is prefered :)

Reply:
> Why is gpio-switch prefered over gpio-export?

I think @mkresin said it best in his post:
https://github.com/openwrt/openwrt/pull/1366#issuecomment-424834448

But this topic comes up on the ML as well.

Reply:
Thanks!!